x-pack/plugins/security/server/authorization/api_authorization.ts

```diff
@@ -47,26 +57,61 @@ export function initAPIAuthorization(
 
       const authz = security.authz as AuthzEnabled;
 
-      const requestedPrivileges = authz.requiredPrivileges.flatMap((privilegeEntry) => {
-        if (typeof privilegeEntry === 'object') {
-          return [...(privilegeEntry.allRequired ?? []), ...(privilegeEntry.anyRequired ?? [])];
+      const { requestedPrivileges, requestedReservedPrivileges } = authz.requiredPrivileges.reduce(
+        (acc, privilegeEntry) => {
+          const privileges =
+            typeof privilegeEntry === 'object'
+              ? [...(privilegeEntry.allRequired ?? []), ...(privilegeEntry.anyRequired ?? [])]
+              : [privilegeEntry];
+
+          for (const privilege of privileges) {
+            if (isReservedPrivilegeSet(privilege)) {
+              acc.requestedReservedPrivileges.push(privilege);
+            } else {
+              acc.requestedPrivileges.push(privilege);
+            }
+          }
+
+          return acc;
+        },
+        {
+          requestedPrivileges: [] as string[],
+          requestedReservedPrivileges: [] as string[],
         }
+      );
 
-        return privilegeEntry;
-      });
-
-      const apiActions = requestedPrivileges.map((permission) => actions.api.get(permission));
       const checkPrivileges = checkPrivilegesDynamicallyWithRequest(request);
-      const checkPrivilegesResponse = await checkPrivileges({ kibana: apiActions });
+      const checkPrivilegesIfNotEmpty = async () => {
+        if (requestedPrivileges.length === 0) {
+          return;
+        }
+
+        const apiActions = requestedPrivileges.map((permission) => actions.api.get(permission));
+
+        return await checkPrivileges({ kibana: apiActions });
+      };
 
       const privilegeToApiOperation = (privilege: string) =>
         privilege.replace(API_OPERATION_PREFIX, '');
+
+      const checkPrivilegesResponse = await checkPrivilegesIfNotEmpty();
```

Comment:
>[!TIP]
> Optional, I'm wondering if something like that would be a bit more easier to "parse" since it clearer when we make the call to `hasPrivileges`:
```ts
      const kibanaPrivileges: Record<string, boolean> = {};
      if (requestedPrivileges.length > 0) {
        const checkPrivilegesResponse = await checkPrivileges({
          kibana: requestedPrivileges.map((permission) => actions.api.get(permission)),
        });

        for (const kbPrivilege of checkPrivilegesResponse.privileges.kibana) {
          kibanaPrivileges[privilegeToApiOperation(kbPrivilege.privilege)] = kbPrivilege.authorized;
        }
      }
```

Reply:
Addressed in 97daaf8b45ba9f6f3992fea1c643b6ef69de9c12